app/controllers/search_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SearchController < ApplicationController
+  def search
+    @drink_results = drink_search(params[:search])
+    @ingredient_results = ingredient_search(params[:search])
+  end
+
+  def drink_name_search(search)
+    Drink.where("name LIKE ?", "%#{search}%")
+  end
+  
+  def ingredient_search(search)
+    Ingredient.where("name LIKE ?", "%#{search}")
```

Comment:
Looks like there's a missing % on the end of this one which causes search terms like "lime" to not return Lime Juice (this would be a good test case)